### PR TITLE
Consolidate status codes

### PR DESCRIPTION
**Pull request by cristaloleg**

### CODE REVIEW

File: frame.go [hunk 1]
```diff
@@ -204,31 +204,29 @@ type StatusCode int
 // These codes were retrieved from:
 // https://www.iana.org/assignments/websocket/websocket.xhtml#close-code-number
 const (
-	StatusNormalClosure StatusCode = 1000 + iota
```
Comment (nhooyr) [inline frame.go]: This is 👍  since iota does make it harder to tell what number each status code corresponds.

File: frame.go [hunk 2]
```diff
@@ -272,7 +270,7 @@ func parseClosePayload(p []byte) (CloseError, error) {
 // and https://tools.ietf.org/html/rfc6455#section-7.4.1
 func validWireCloseCode(code StatusCode) bool {
 	switch code {
-	case 1004, StatusNoStatusRcvd, StatusAbnormalClosure, StatusTLSHandshake:
+	case StatusReserved, StatusNoStatusRcvd, StatusAbnormalClosure, StatusTLSHandshake:
```
Comment (nhooyr) [inline frame.go]: Can you unexport status reserved?

Comment (cristaloleg) [reply frame.go]: hm, agree, should be better

**Comment by nhooyr** [comment]
@cristaloleg Could you change the second commit to start with a capital? Then this should be good for merge.

**Pull request closed without merging**